ext/nnstreamer/tensor_filter/tensor_filter_python3.cc

```diff
@@ -612,272 +533,175 @@ PYCore::run (const GstTensorMemory *input, GstTensorMemory *output)
-    return -ENOENT;
-  }
-
-  PyGILState_STATE gstate = PyGILState_Ensure ();
-  int ret = core->getOutputTensorDim (info);
-  PyGILState_Release (gstate);
-  return ret;
+TensorFilterPython::TensorFilterPython () {
+  core = nullptr;
+  if (!Py_IsInitialized ())
+    throw std::runtime_error ("Python is not initialize.");
 }
 
 /**
- * @brief Free privateData and move on.
+ * @brief Method to get an empty object
  */
-static void
-py_close (const GstTensorFilterProperties *prop, void **private_data)
-{
-  PYCore *core = static_cast<PYCore *> (*private_data);
-  UNUSED (prop);
-
-  g_return_if_fail (core != NULL);
-  PyGILState_STATE gstate = PyGILState_Ensure ();
-  delete core;
-  PyGILState_Release (gstate);
-
-  *private_data = NULL;
+tensor_filter_subplugin & TensorFilterPython::getEmptyInstance () {
+  return *(new TensorFilterPython ());
 }
 
 /**
- * @brief Load python model
- * @param prop: property of tensor_filter instance
- * @param private_data : python plugin's private data
- * @return 0 if successfully loaded. 1 if skipped (already loaded).
- *        -1 if the object construction is failed.
- *        -2 if the object initialization if failed
+ * @brief Configure Python instance
  */
-static int
-py_loadScriptFile (const GstTensorFilterProperties *prop, void **private_data)
+void TensorFilterPython::configure_instance (const GstTensorFilterProperties *
+    prop)
 {
-  PYCore *core;
   const gchar *script_path;
 
   if (prop->num_models != 1)
-    return -1;
+    return;
 
   /**
    * prop->model_files[0] contains the path of a python script
    * prop->custom contains its arguments seperated by ' '
    */
-  core = static_cast<PYCore *> (*private_data);
   script_path = prop->model_files[0];
 
-  if (core != NULL) {
-    if (g_strcmp0 (script_path, core->getScriptPath ()) == 0)
-      return 1; /* skipped */
+  if (core != nullptr) {
+    if (g_strcmp0 (script_path, core->getScriptPath ()) == 0) {
+      return; /* skipped */
+    }
 
-    py_close (prop, private_data);
+    PyGILState_STATE gstate = PyGILState_Ensure ();
+    delete core;
+    PyGILState_Release (gstate);
   }
 
-  /* init null */
-  *private_data = NULL;
-
   PyGILState_STATE gstate = PyGILState_Ensure ();
   core = new PYCore (script_path, prop->custom_properties);
-  if (core == NULL) {
+  if (core == nullptr) {
     g_printerr ("Failed to allocate memory for filter subplugin: Python\n");
     PyGILState_Release (gstate);
-    return -1;
+    return;
   }
 
   if (core->init (prop) != 0) {
     delete core;
     g_printerr ("failed to initailize the object: Python\n");
     PyGILState_Release (gstate);
-    return -2;
+    throw std::runtime_error ("Python is not initialize");
   }
 
   /** check methods in python script */
   if (core->getCbType () != CB_SETDIM && core->getCbType () != CB_GETDIM) {
     delete core;
     g_printerr ("Wrong callback type\n");
     PyGILState_Release (gstate);
-    return -2;
+    return;
```

Comment:
Each if-statement in these lines looks like doing similar things as follows;
- g_printerr
- PyGILState_Release
- return or throw

Could you apply code-refactoring for these lines to reduce code duplication?

Reply:
Used goto after g_printerr!
Thank you :)